app/models/miq_schedule_worker/runner.rb

```diff
@@ -439,15 +439,15 @@ def rufus_add_monthly_schedule(options)
     # Treat months differently since rufus doesn't support :schedule_every with X.months type of options
     sch = MiqSchedule.find(schedule_id)
     next_run = sch.next_interval_time
-    @schedules[:scheduler] << @user_scheduler.send(method, next_run, options) do |rufus_job|
+    @schedules[:scheduler] << @user_scheduler.send(method, next_run, options.dup) do |rufus_job|
       enqueue [:miq_schedule_queue_scheduled_work, schedule_id, rufus_job]
     end
 
     # Schedule every X months for up to 5 years in the future
     remaining_months = ((5 * 12) / months) - 1
     remaining_months.times do
       next_run += months.months
-      @schedules[:scheduler] << @user_scheduler.send(method, next_run, options) do |rufus_job|
+      @schedules[:scheduler] << @user_scheduler.send(method, next_run, options.dup) do |rufus_job|
```

Comment:
@imtayadeway it's not obvious at first glance but they parse the time string using Chronic starting in version 3.0.8.  Because this is slow, they cache the result by mutating OUR options hash we supplied.  In our code here, we pass the exact same options hash for each call to ` @user_scheduler.schedule_at` (send(method)).  

Therefore, the second, third, fourth, etc. call to `.send` below is passing an options hash that has been mutated with the result of the first time parse.  Therefore, we try to create 1 schedule for each month for 5 years worth with 60 different times, but end up getting 60 schedules with the same exact run at time, the first time!